From 8e1179a901adb11b8b25e85fdec2d624e397a6d6 Mon Sep 17 00:00:00 2001
From: Christian Ege <k4230r6@gmail.com>
Date: Tue, 26 Jul 2016 19:36:51 +0200
Subject: [PATCH] Removed dts folder from device tree file name

This is due to the fact that the meta-fsl-arm layer does provide an
SD-card image class which do not put the dtb files in a separate subfolder

Signed-off-by: Christian Ege <k4230r6@gmail.com>
---
 board/udoo/a62/mx6qd_a62.c | 7 +++----
 board/udoo/udoo_neo/neo.c  | 7 +++----
 board/udoo/udoo_qd/udoo.c  | 7 +++----
 3 files changed, 9 insertions(+), 12 deletions(-)

diff --git a/board/udoo/a62/mx6qd_a62.c b/board/udoo/a62/mx6qd_a62.c
index ca264ab..944705e 100644
--- a/board/udoo/a62/mx6qd_a62.c
+++ b/board/udoo/a62/mx6qd_a62.c
@@ -592,18 +592,17 @@ int do_a62init(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 		}
 	}
 	
-	char* dir_part = "dts";
+	char fdt_file[100];
 	char* customdtb = getenv("use_custom_dtb");
+	sprintf(fdt_file, "%s%s.dtb", modelfdt, video_part);
 	if (customdtb) {
 		customdtb = trim(customdtb);
 		if (strcmp(customdtb, "true") == 0 || strcmp(customdtb, "yes") == 0 || strcmp(customdtb, "enabled") == 0) {
 			dir_part = "dts-overlay";
+			sprintf(fdt_file, "%s/%s%s.dtb", dir_part, modelfdt, video_part);
 		}
 	}
 	
-	char fdt_file[100];
-	sprintf(fdt_file, "%s/%s%s.dtb", dir_part, modelfdt, video_part);
-	
 	printf("Device Tree: %s\n", fdt_file);
 	setenv("fdt_file", fdt_file);
 	
diff --git a/board/udoo/udoo_neo/neo.c b/board/udoo/udoo_neo/neo.c
index 7e46a59..2638640 100644
--- a/board/udoo/udoo_neo/neo.c
+++ b/board/udoo/udoo_neo/neo.c
@@ -1117,18 +1117,17 @@ int do_udooinit(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 		}
 	}
 	
-	char* dir_part = "dts";
+	char fdt_file[100];
 	char* customdtb = getenv("use_custom_dtb");
+	sprintf(fdt_file, "%s%s%s.dtb", modelfdt, video_part, m4_part);
 	if (customdtb) {
 		customdtb = trim(customdtb);
 		if (strcmp(customdtb, "true") == 0 || strcmp(customdtb, "yes") == 0 || strcmp(customdtb, "enabled") == 0) {
 			dir_part = "dts-overlay";
+			sprintf(fdt_file, "%s/%s%s%s.dtb", dir_part, modelfdt, video_part, m4_part);
 		}
 	}
 	
-	char fdt_file[100];
-	sprintf(fdt_file, "%s/%s%s%s.dtb", dir_part, modelfdt, video_part, m4_part);
-	
 	printf("Device Tree: %s\n", fdt_file);
 	setenv("fdt_file", fdt_file);
 	
diff --git a/board/udoo/udoo_qd/udoo.c b/board/udoo/udoo_qd/udoo.c
index 1276e26..cccb5ba 100644
--- a/board/udoo/udoo_qd/udoo.c
+++ b/board/udoo/udoo_qd/udoo.c
@@ -439,18 +439,17 @@ int do_udooinit(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 		}
 	}
 	
-	char* dir_part = "dts";
+	char fdt_file[100];
 	char* customdtb = getenv("use_custom_dtb");
+	sprintf(fdt_file, "%s%s.dtb", modelfdt, video_part);
 	if (customdtb) {
 		customdtb = trim(customdtb);
 		if (strcmp(customdtb, "true") == 0 || strcmp(customdtb, "yes") == 0 || strcmp(customdtb, "enabled") == 0) {
 			dir_part = "dts-overlay";
+			sprintf(fdt_file, "%s/%s%s.dtb", dir_part, modelfdt, m4_part);
 		}
 	}
 	
-	char fdt_file[100];
-	sprintf(fdt_file, "%s/%s%s.dtb", dir_part, modelfdt, video_part);
-	
 	printf("Device Tree: %s\n", fdt_file);
 	setenv("fdt_file", fdt_file);
 	
-- 
2.7.4

